### PR TITLE
remove unused regions from UnsafeOrdering

### PR DESCRIPTION
**Pull request by patrick-schultz**

### CODE REVIEW

File: hail/src/main/scala/is/hail/sparkextras/ContextRDD.scala [hunk 1]
```diff
@@ -265,6 +265,32 @@ class ContextRDD[C <: AutoCloseable, T: ClassTag](
       preservesPartitioning),
     mkc)
 
+  def cmapPartitionsWithContext[U: ClassTag](f: (C, (C) => Iterator[T]) => Iterator[U]): ContextRDD[C, U] = {
```
Comment (tpoterba) [inline hail/src/main/scala/is/hail/sparkextras/ContextRDD.scala]: how do these changes fit in?

Comment (patrick-schultz) [reply hail/src/main/scala/is/hail/sparkextras/ContextRDD.scala]: They don't. Sorry, I broke this off of a bigger change trying to straighten out the region management in RVD, and these accidentally slipped in. Deleting.

**Comment by patrick-schultz** [comment]
That was nasty to debug. For some reason, Scala has both `Ordering.on` (on the `Ordering` class) and `Ordering.by` (on the companion object) to translate an ordering on `U` to an ordering on `T` along a function `f: T => U`.

`on` defines `compare`:
```
  def on[U](f: U => T): Ordering[U] = new Ordering[U] {
    def compare(x: U, y: U) = outer.compare(f(x), f(y))
  }
```
but `by` mysteriously bases the new ordering only on the original `lt`:
```
  def by[T, S](f: T => S)(implicit ord: Ordering[S]): Ordering[T] =
    fromLessThan((x, y) => ord.lt(f(x), f(y)))

  def fromLessThan[T](cmp: (T, T) => Boolean): Ordering[T] = new Ordering[T] {
    def compare(x: T, y: T) = if (cmp(x, y)) -1 else if (cmp(y, x)) 1 else 0
    // overrides to avoid multiple comparisons
    override def lt(x: T, y: T): Boolean = cmp(x, y)
    override def gt(x: T, y: T): Boolean = cmp(y, x)
    override def gteq(x: T, y: T): Boolean = !cmp(x, y)
    override def lteq(x: T, y: T): Boolean = !cmp(y, x)
  }
```

**Comment by tpoterba** [comment]
wow, that's rough.

**Pull request merged**